tasks/save-collectors-results/tests/test-save-collectors-results-no-results.yaml

```diff
@@ -0,0 +1,48 @@
+---
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: test-save-collectors-results-no-results
+spec:
+  description: |
+    Run the save-collectors-results task where there are no files in the results dir.
+    The pipeline should succeed.
+  workspaces:
+    - name: tests-workspace
+  tasks:
+    - name: setup
+      taskSpec:
+        steps:
+          - name: setup
+            image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+
+              cat > release << EOF
+              apiVersion: appstudio.redhat.com/v1alpha1
+              kind: Release
+              metadata:
+                name: release-cr-no-results
+                namespace: default
+              spec:
+                snapshot: foo
+                releasePlan: foo
+              EOF
+              kubectl apply -f release
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+    - name: run-task
```

Comment:
You should also add a check result task and check that the status was added. It will be empty, but it will still be there, because the task will still run the kubectl command.

Reply:
The field is optional so it wouldn't be there.

Reply:
Which field? This command will still run:
```
kubectl --warnings-as-errors=true patch "$(params.resourceType)" -n "$namespace" "$name" \
          --type=merge --subresource status --patch "status: {'$(params.statusKey)':${RESULTS_JSON}}"
```
And `$RESULTS_JSON` will be just `{}`.

Reply:
This will set `status.collectors` to `{}`. Given that in the Release resource that field is optional, checking the release will not show it as it's set to the nil value.

Reply:
Ah, I see, thanks.